stdlib/public/core/StringCreate.swift

```diff
@@ -19,33 +19,54 @@ internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
   //
   // TODO(String performance): SIMD-ize
   //
-  let ptr = input.baseAddress._unsafelyUnwrappedUnchecked
-  var i = 0
-
   let count = input.count
-  let stride = MemoryLayout<UInt>.stride
-  let address = Int(bitPattern: ptr)
-
-  let wordASCIIMask = UInt(truncatingIfNeeded: 0x8080_8080_8080_8080 as UInt64)
-  let byteASCIIMask = UInt8(truncatingIfNeeded: wordASCIIMask)
+  var ptr = UnsafeRawPointer(input.baseAddress._unsafelyUnwrappedUnchecked)
+  var i = 0
 
-  while (address &+ i) % stride != 0 && i < count {
-    guard ptr[i] & byteASCIIMask == 0 else { return false }
-    i &+= 1
+  let asciiMask64 = 0x8080_8080_8080_8080 as UInt64
+  let asciiMask32 = UInt32(truncatingIfNeeded: asciiMask64)
+  let asciiMask16 = UInt16(truncatingIfNeeded: asciiMask64)
+  let asciiMask8 = UInt8(truncatingIfNeeded: asciiMask64)
+  
+  let end128 = ptr + count & ~(MemoryLayout<(UInt64, UInt64)>.stride &- 1)
+  let end64 = ptr + count & ~(MemoryLayout<UInt64>.stride &- 1)
+  let end32 = ptr + count & ~(MemoryLayout<UInt32>.stride &- 1)
+  let end16 = ptr + count & ~(MemoryLayout<UInt16>.stride &- 1)
+  let end = ptr + count
+
+  
+  while ptr < end128 {
```

Comment:
Could it be faster to determine the number of loops you're doing up front, since you already have `count` calculated?

Reply:
If the count were known up front that could help the loop unroller but in this case I don't think it can help us. Not incrementing the pointer directly means we need a load with offset, and I don't think there's a version of `ldp` that takes a register-offset pair. Two loads and an add is quite enough for one instruction 😅